samples/net/sockets/echo_async_select/sample.yaml

```diff
@@ -7,7 +7,7 @@ common:
   tags:
     - net
     - socket
-  filter: TOOLCHAIN_HAS_NEWLIB == 1
+  filter: CONFIG_FULL_LIBC_SUPPORTED
```

Comment:
Side-tangent: the extra_configs in sample.net.sockets.echo_async_select.posix is a bit bogus as CONFIG_POSIX_API=y is set in the prj.conf. The whole thing needs a filter on ` !CONFIG_NATIVE_APPLICATION`